network/builder/src/builder.rs

```diff
@@ -166,42 +163,39 @@ impl NetworkBuilder {
             .expect("Seed peer addresses must be well-formed");
         let seed_peers = config.seed_peers.clone();
 
-        if config.mutual_authentication {
-            let network_peers = config.network_peers.clone();
-            let trusted_peers = if role == RoleType::Validator {
-                // for validators, trusted_peers is empty will be populated from consensus
-                HashMap::new()
-            } else {
-                network_peers
-            };
-
-            info!(
-                "network setup: role: {}, seed_peers: {:?}, trusted_peers: {:?}",
-                role, seed_peers, trusted_peers,
-            );
+        info!(
+            "network setup: role: {}, seed_peers: {:?}, trusted_peers: {:?}",
+            role, seed_peers, trusted_peers,
+        );
 
-            network_builder
-                .trusted_peers(trusted_peers)
-                .seed_peers(seed_peers)
-                .connectivity_check_interval_ms(config.connectivity_check_interval_ms)
-                .add_connectivity_manager();
-        } else {
-            // Enforce the outgoing connection (dialer) verifies the identity of the listener (server)
-            if config.discovery_method == DiscoveryMethod::Onchain || !seed_peers.is_empty() {
-                network_builder
-                    .seed_peers(seed_peers)
-                    .add_connectivity_manager();
-            }
-        }
+        // Setup the ConnectivityManager.  Do not setup a ConnectivityManager iff
+        //   1.  no mutual authentication
+        //   2.  DiscoveryMethod::Gossip
+        //   3.  no seed_peers
+        // TODO:  Why not add ConnectivityManager always?
```

Comment:
tech debt, it will kill connections if they aren't in the trusted peers test... which makes no sense for server-only auth...

Reply:
This seems very .... bad to me.  I think we should fix that behavior so connectivitymanager can be added always.

Reply:
I think it comes more to the problem of the general networking model we're using at this point, where we expect it to be a p2p network, and that the connectivity manager only connects to trusted versions of those.

I think we need to look forward into refactoring this to allow for that situation, but it's not made yet for this.

Reply:
I would expect that ConnectivityManager is (a) reconnecting to the approved peers and (b) only accepting lincoming connections from approved peers.

I would hope that the difference in the scenario you hint at is simply that the list of approvied peers changes.  But I dont have enough context on the intent.

It's also extraordinarily odd to me that the existence of seed_peers is the deciding factor as to whether or not to instantiate the connectivitymanager.